### DISCUSSION TITLE
Feature Request: Possibility to disable the scroll lock behavior on Modals

### DISCUSSION THREAD

**Discussion by thibaut-trouve** [tailwindlabs/headlessui#478] [Ideas]
### **Feature:**
At the moment it's not possible to disable the scroll lock behavior when a Dialog is opened.

We could add a prop "disableScrollLock" on the Dialog component, like on Material-UI's Modal: https://material-ui.com/api/modal/ that would prevent from blocking the scroll by adding a condition in this file when disableScrollLock === true: https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/dialog/dialog.tsx#L181

**Comment by claudesortwell** [comment]
I just opened another discussion with the same request:
https://github.com/tailwindlabs/headlessui/discussions/510
> I think headless UI is amazing, but I feel like the the ability to scroll on the outside of a Dialog is very useful in some cases (e.g. slide over) and with such a barebone UI library, I would think it would not be forced or at least controllable.
> 
> I was wondering if their was some reason of not exposing the ability to adjust or disable the scroll lock on a dialog?
> 
> If not I'm more than happy to open a pull request for it.

I guess I will open a pull request with this ability and see if the team will accept it.
